node/src/beacon/router.rs

```diff
@@ -76,9 +76,17 @@ impl<N: Network> Inbound<N> for Beacon<N> {
             trace!("Skipping 'UnconfirmedSolution' from '{peer_ip}'");
         } else {
             // Add the unconfirmed solution to the memory pool.
-            if let Err(error) = self.ledger.consensus().write().add_unconfirmed_solution(&solution) {
-                trace!("[UnconfirmedSolution] {error}");
-                return true; // Maintain the connection.
+            match self.ledger.consensus().try_write() {
+                Some(mut ledger) => {
+                    if let Err(error) = ledger.add_unconfirmed_solution(&solution) {
+                        trace!("[UnconfirmedSolution] {error}");
+                        return true; // Maintain the connection.
+                    }
+                }
+                None => {
+                    trace!("Ignoring an unconfirmed solution (already creating a block).");
+                    return true;
+                }
```

Comment:
The unconfirmed solutions are valid over an epoch (many blocks), so we shouldn't be ignoring them here. They are still valid as long as you are within the epoch block range (currently 256 blocks).